bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContentAssistRequest.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuł and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pakuła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Describes the state that the content assistant is in when completing proposals.
+ * <p>
+ * Clients may use this class.
+ * </p>
+ *
+ * @since 3.24
+ * @see IContentAssistProcessorExtension2
+ * @noinstantiate This class is not intended to be instantiated by clients
+ */
+public final class ContentAssistRequest {
```

Comment:
An interface would be more suitable (and flexible) here I think

Reply:
Sure

Reply:
Done